pkg/agent/storage/azure.go

```diff
@@ -0,0 +1,217 @@
+	GetAllObjects() ([]string, error)
+	Download(blobsNames []string) error
+	DownloadSingle(blobName string) error
+}
+
+var _ Provider = (*AzureProvider)(nil)
+
+func (a *AzureProvider) DownloadModel(modelDir string, modelName string, storageUri string) error {
+	// Parse Azure URI
+	uriParts, err := parseAzureUri(storageUri)
+	if err != nil {
+		return fmt.Errorf("failed to parse Azure URI: %v", err)
+	}
+
+	if a.Downloader == nil {
+		log.Info("Initializing Azure downloader")
+		a.Downloader, err = newDefaultDownloader(uriParts.serviceUrl)
+		if err != nil {
+			return fmt.Errorf("failed to initialize Azure downloader: %v", err)
+		}
+	}
+
+	a.Downloader.SetDownloaderValues(modelDir, modelName, uriParts.containerName, uriParts.prefix)
+	blobs, err := a.Downloader.GetAllObjects()
+	if err != nil {
+		return fmt.Errorf("failed to get blobs: %v", err)
+	}
+	if len(blobs) == 0 {
+		return fmt.Errorf("no blobs found")
+	}
+	if err := a.Downloader.Download(blobs); err != nil {
+		return fmt.Errorf("failed to download blobs: %v", err)
+	}
+	return nil
+}
+
+func newDefaultDownloader(serviceUrl string) (AzureDownloader, error) {
+	// Initialize AzureBlobClient
+	var err error
+	var client *azblob.Client
+	if _, ok := os.LookupEnv(azure.AzureClientSecret); ok {
+		// Load Azure Credentials
+		credential, err := azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load Azure credentials: %v", err)
+		}
+		client, err = azblob.NewClient(serviceUrl, credential, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create Azure client with credentials: %v", err)
+		}
+	} else {
+		client, err = azblob.NewClientWithNoCredential(serviceUrl, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create Azure client with no credentials: %v", err)
+		}
+	}
+
+	return &defaultAzureDownloader{
+		client: client,
+	}, nil
+}
+
+// ParseAzureUri is a method that parses the Azure URI and sets the azureUriParts.
+// The Azure URI should follow this specific format: azure://<serviceUrl>/<containerName>/<blob>
+//
+// Here's a breakdown of the URI components:
+//   - <serviceUrl>: The base URL of your Azure storage account. For example, 'myStorageAccount.blob.core.windows.net'
+//   - <containerName>: The specific containerName in the Azure storage account. For example, 'myContainer'
+//   - <blob>: The specific blob within the containerName. This can include a path with directories. For example, 'this/is/virtualDir/fileName'
+//
+// Putting it all together, an example Azure storageURI would look like this:
+// 'azure://myStorageAccount.blob.core.windows.net/myContainer/this/is/virtualDir/fileName'
+//
+// In this example,
+//   - serviceUrl would be 'myStorageAccount.blob.core.windows.net'
+//   - containerName would be 'myContainer'
+//   - prefix (the path to the blob within the containerName) would be 'this/is/virtualDir/fileName'.
+func parseAzureUri(uri string) (azureUriParts, error) {
```

Comment:
Add some units for this?

Reply:
Thanks for suggestion, I add some unit tests as part of commit c89926fac0033569208134a5c733037b752b59ee